sanitize.go

```diff
@@ -0,0 +1,50 @@
+package inputsanitizer
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const sanitize_replacement_string string = "REDACTED"
+
+var sensitiveFieldNames = []string{
+	`account[_|/-|\s](.[\w\s_/-]+)*key`,
+	`api[_|/-|\s](.[\w\s_/-]+)*key`,
+	`auth[_|/-|\s](.[\w\s_/-]+)*key`,
+	`azurerm-account-key`,
+	`client[_|/-](.[\w\s_/-]+)*key`,
+	`contrasena`,
+	`contraseña`,
+	`database[_|/-|\s](.[\w\s_/-]+)*key`,
+	`database[_|/-|\s](.[\w\s_/-]+)*pass`,
+	`database[_|/-|\s](.[\w\s_/-]+)*password`,
+	`db[_|/-|\s](.[\w\s_/-]+)*key`,
+	`db[_|/-|\s](.[\w\s_/-]+)*pass`,
+	`db[_|/-|\s](.[\w\s_/-]+)*password`,
+	`fetch-tfstate-headers`,
+	`key[_|/-|\s](.[\w\s_/-]+)*pass`,
+	`key[_|/-|\s](.[\w\s_/-]+)*password`,
+	`passwd`,
+	`password`,
+	`priv[_|/-|\s](.[\w\s_/-]+)*key`,
+	`private[_|/-|\s](.[\w\s_/-]+)*key`,
+	`pwd`,
+	`secret`,
+	`service[_|/-|\s](.[\w\s_/-]+)*key`,
+	`tfc-token`,
+	`username`,
+}
+
+func SanitizeValuesByKey(content []byte) ([]byte, error) {
+	for i := range sensitiveFieldNames {
+		filter := sensitiveFieldNames[i]
+		r, err := regexp.Compile(fmt.Sprintf(`(?:\"|\')(?P<key>(%s)+)(?:\"|\')(?:\:\s*)(?:\"|\')?(?P<value>[\w\s-\[\]]*)(?:\"|\')?`, filter))
```

Comment:
we can statically compile these https://github.com/di-graph/digraph-repo/blob/main/go_svcs/pkgs/response_generators/utils.go#L23

it might make this more performant if the `sensitiveFieldNames` is a list of precompiled regexes.